packages/edit-site/src/components/dataviews/view-grid.js

```diff
@@ -25,41 +32,59 @@ export function ViewGrid( { data, fields, view, actions, getItemId } ) {
 	);
 	return (
 		<Grid gap={ 8 } columns={ 2 } alignment="top">
-			{ data.map( ( item, index ) => {
-				return (
-					<VStack key={ getItemId?.( item ) || index }>
-						<div className="dataviews-view-grid__media">
-							{ mediaField?.render( { item, view } ) || (
-								<Placeholder
-									withIllustration
-									style={ {
-										width: '100%',
-										minHeight: '200px',
-									} }
-								/>
-							) }
-						</div>
-
-						<HStack justify="space-between" alignment="top">
-							<FlexBlock>
-								<VStack>
-									{ visibleFields.map( ( field ) => (
-										<div key={ field.id }>
-											{ field.render( { item, view } ) }
-										</div>
-									) ) }
-								</VStack>
-							</FlexBlock>
-							<FlexBlock>
-								<ItemActions
-									item={ item }
-									actions={ actions }
-								/>
-							</FlexBlock>
-						</HStack>
-					</VStack>
-				);
-			} ) }
+			{ data.map( ( item, index ) => (
+				<ViewGridItem
+					key={ getItemId?.( item ) || index }
+					mediaField={ mediaField }
```

Comment:
`mediaField` and `visibleFields` could also be absorved by `ViewGridItem`, so its API/props is only `item`, `view`, and `actions`.

Reply:
We would need to also pass fields. But the main thing for me here was that `mediaField` and `visibleFields` are not dependent on the item, so there is no need to loop the array to get the same result for each item.